File: SonicXBackned/src/main/java/org/example/DataBase/DatabaseConnector.java
package org.example.DataBase;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class DatabaseConnector {

    private static final String JDBC_URL = "jdbc:mysql://localhost:3306/your_database";
    private static final String USERNAME = "root";
    private static final String PASSWORD = "#123456789#";

    private static DatabaseConnector instance;
    private Connection connection;

    private DatabaseConnector() throws SQLException {
        try {
            Class.forName("com.mysql.cj.jdbc.Driver");
            this.connection = DriverManager.getConnection(JDBC_URL, USERNAME, PASSWORD);
            System.out.println("Veritabanına bağlandı!");
        } catch (ClassNotFoundException e) {
            throw new SQLException("JDBC sürücüsü bulunamadı: " + e.getMessage());
        }
    }

    public static DatabaseConnector getInstance() throws SQLException {
        if (instance == null) {
            instance = new DatabaseConnector();
        } else if (instance.getConnection().isClosed()) {
            instance = new DatabaseConnector();
        }
        return instance;
    }

    public Connection getConnection() {
        return connection;
    }

    public void disconnect() {
        if (connection != null) {
            try {
                connection.close();
                System.out.println("Veritabanı bağlantısı kapatıldı.");
            } catch (SQLException e) {
                System.out.println("Veritabanı bağlantısı kapatılırken hata oluştu: " + e.getMessage());
            }
        }
    }
}

File: SonicXBackendC/SonicXBackendC/DataBase/DatabaseConnector.cs
using System;
using MySql.Data.MySqlClient;

namespace SonicXBackendC.DataBase
{
    public class DatabaseConnector
    {
        private static readonly string ConnectionString = "Server=localhost;Port=3306;Database=your_database;Uid=root;Pwd=your_password";
        private static DatabaseConnector instance;
        private MySqlConnection connection;

        private DatabaseConnector()
        {
            try
            {
                connection = new MySqlConnection(ConnectionString);
                connection.Open();
                Console.WriteLine("Veritabanına bağlandı!");
            }
            catch (MySqlException e)
            {
                throw new Exception("MySQL bağlantı hatası: " + e.Message);
            }
        }

        public static DatabaseConnector GetInstance()
        {
            if (instance == null || instance.connection.State == System.Data.ConnectionState.Closed)
            {
                instance = new DatabaseConnector();
            }
            return instance;
        }

        public MySqlConnection GetConnection()
        {
            return connection;
        }

        public void Disconnect()
        {
            if (connection != null)
            {
                try
                {
                    connection.Close();
                    Console.WriteLine("Veritabanı bağlantısı kapatıldı.");
                }
                catch (MySqlException e)
                {
                    Console.WriteLine("Veritabanı bağlantısı kapatılırken hata oluştu: " + e.Message);
                }
            }
        }
    }
}
